modules/tensor_mechanics/include/auxkernels/nodal_patch_recovery/NPRAux.h

```diff
@@ -0,0 +1,28 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "AuxKernel.h"
+
+class NPRBase;
+
+class NPRAux : public AuxKernel
```

Comment:
> unless you think we're going to have 10 or more derived classes specifically from nodal patch recovery

I agree with you, but I'm afraid the answer to this is yes... If we have @dschwen 's `IndexableProperty` I could probably reduce the number of derived classes hence expand this class name.

Reply:
We might have IndexableProperty before Alex' predicate makes it into libmesh (and MOOSE, through a submodule update)

Reply:
Probably since this libmesh update is a brutal one (we will be needing to patch around 50 tests in BISON)

Reply:
I was discussing this with @bwspenc just today. It sounds like we are moving towards having a single material object that extracts all sorts of tensor_mechanics scalar quantities for output purposes (not sure which class is doing that though). So that is another option.

Either way, I think I don't need to add a whole bunch of userobjects to do nodal patch recovery, so I will remove the subdirectory and expand `NPR` in the class names.

Reply:
> Probably since this libmesh update is a brutal one (we will be needing to patch around 50 tests in BISON)

I really hope this brutal update doesn't make the bison failure in #16816 worse...